src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -60,6 +60,32 @@ class ChannelInputStream extends InputStream {
         this.ch = ch;
     }
 
+    /*
+     * Reads bytes in DEFAULT_BUFFER_SIZE chunks until all requested bytes
+     * have been read or end-of-stream is reached.
+     * If the channel is selectable then it must be configured blocking.
+     */
+    private int readFully(ByteBuffer bb) throws IOException {
```

Comment:
InputStream.read methods are not "read fully", it will cause the read to hang, say where you read some bytes and block indefinitely to fill the user's buffer.

Reply:
I was ambivalent about that name as well. Changed in a9af89cff869c441c6dd04f635a4d93e7757f2fa.